charts/tekton-pipelines/templates/common/tasks/kustomize.yaml

```diff
@@ -67,7 +67,15 @@ spec:
           CONFIG=$(realpath .yamlfix.toml)
         fi
 
-        find apps -iname "*.yaml" -exec yamlfix --config-file=$CONFIG {} \;
+        find $(params.kustomize_overlay_path) -iname "*.yaml" -exec yamlfix --config-file=$CONFIG {} \;
+
+    - name: kubeval
+      image: {{ .Values.images.kubeval | default "public.ecr.aws/saritasa/kubeval:latest"}}
+      imagePullPolicy: {{ .Values.imagePullPolicy }}
+      workingDir: $(resources.inputs.kubernetes-repo.path)
+      script: |
+        export KUBEVAL_SCHEMA_LOCATION=https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master
+        kustomize build $(params.kustomize_overlay_path) | kubeval -f -
```

Comment:
I checked - if `kubeval` step fails, `git-push` is not executed: [ex.](https://tekton.saritasa.rocks/#/namespaces/ci-experiments/pipelineruns/saritasa-ci-experiments-php-backend-dev-build-pipeline-run8c8wj?pipelineTask=kustomize&step=kubeval) here it fails for another reason, but I've seen result with schema validation fail - it works the same
![2023-02-20_20-25_mmngDkvi](https://user-images.githubusercontent.com/3766033/220146085-5e9b0126-9fbb-4325-ba49-53d94cfe8929.png)

Reply:
@dmitry-mightydevops I didn't implement what you asked for - rollback if validation fails. 
I still think that it is dangerous to patch whole `apps` forlder (including `base`). For example, we patch `base` and we validate successfully against Dev cluster `v1.24`. But `prod` overlay still may fail against another cluster version, ex. `v1.22`, but we cannot detect it, since we cannot retrive prod cluster version. So I'd beter just leave responsibility for `base` forlder modifications to `pre-commit` tools and committer, rather than have too many "magic" in deployment process.